src/plugins/vis_types/timeseries/public/application/components/vis_editor_visualization.js

```diff
@@ -194,8 +208,27 @@ class VisEditorVisualizationUI extends Component {
           data-shared-items-container
           data-title={title}
           data-description={description}
-          ref={this._visEl}
-        />
+        >
+          <ReactEmbeddableRenderer
+            type={VISUALIZE_EMBEDDABLE_TYPE}
+            state={{
+              rawState: {
+                id: '',
+                savedVis: this.getSavedVis(),
+                title: title,
+                description: description,
+              },
+              references: [],
+            }}
+            parentApi={this._parentApi}
```

Comment:
This pattern really bugs me. `unifiedSearch` properties are expected to be passed in through a `parentApi`, and they're supposed to be in the form of PublishingSubjects. But in this case (and with DefaultEditor) the current published values of each property is just a React prop. It makes sense from a React point of view to pass these in as props.

We can't, though, because the framework is expecting Observables. So instead we have to kind of break the React paradigm in order to shoehorn these props into some RxJS logic that they don't really want to be in.

Is there any better way to handle this?

Reply:
> Is there any better way to handle this?

No. We want to avoid having multiple ways of doing things.

You could opt to create a component that takes timeRange, filters, and query as props and then encapsulates converting these to observables and passing via the parent API. Maps embeddable does this with the [MapRenderer](https://github.com/elastic/kibana/blob/ca267e03bfa293bd41bbf7ce6853b970491fc5e0/x-pack/plugins/maps/public/react_embeddable/map_renderer.tsx) if you want to use this as an example.

Reply:
I'd say this frustration is warranted. 

@nreese's suggestion may help - additionally you could use `usePublishingSubject` to handle the  translation from `Reactive variable` -> `PublishingSubject` for you. I understand that neither of these solutions remove the translation requirement, they just make it a little easier.

I think the fundamental problem here is a mixup in the source of truth. Ideally if you were building an editor for a React Embeddable **from scratch** all state would be stored with RXJS. That said, it seems like changing the visualize architecture to be fully RXJS based is too large a lift for this PR, so you might be stuck translating for a while.

Ultimately, this is making me think that we shouldn't recommend folks use Embeddables in their standalone editors.

Reply:
We could also create a hook that could turn filters, timeRange, and query into a search API to turn this into a single line of code for consumers.

```
const searchApi = useSearchApi(filters, timeRange, query);
```

Reply:
That's not a bad idea. Kinda the opposite of `useFetchContext`

Reply:
right I can add it to this PR https://github.com/elastic/kibana/pull/183772

> Ultimately, this is making me think that we shouldn't recommend folks use Embeddables in their standalone editors.

I think this highlights the [embeddable key principles and best practices](https://github.com/elastic/kibana/blob/main/src/plugins/embeddable/README.md) to only use embeddables when you need a component that can render in dashboard. For all other cases, expose a react Component. The problems that embeddables are trying to solve make them more difficult to work with then standard react components.

Reply:
I have added `useSearchApi` hook in https://github.com/elastic/kibana/pull/183772. You can check it out at https://github.com/elastic/kibana/blob/95428bd83f835ae8edefe833159fd2c60bd6da4a/packages/presentation/presentation_publishing/interfaces/fetch/publishes_unified_search.ts

Once that PR merges, you can just use that hook and this should save you a few lines and make working with embeddables more managable.